src/basics.md

```diff
@@ -1101,6 +1101,68 @@ fn main() {
 }
 ```
 
+[ex-error-chain-backtrace]: #ex-error-chain-backtrace
+<a name="ex-error-chain-backtrace"></a>
+## Obtain backtrace of complex error scenarios
+
+[![error-chain-badge]][error-chain] [![cat-rust-patterns-badge]][cat-rust-patterns]
+
+Extend errors by appending new errors via [`chain_err`]. It provides a
+better context to understand an error in a form of a backtrace. The
+below recipes attemps to deserialize the value `256` into a `u8`. An
+error will bubble up from Serde then csv and finaly up to the user
+code.
+
+```rust
+# extern crate csv;
+#[macro_use]
+extern crate error_chain;
+# #[macro_use]
+# extern crate serde_derive;
+#
+# error_chain! {
+#     foreign_links {
+#         Reader(csv::Error);
+#     }
+# }
+
+#[derive(Debug, Deserialize)]
+struct Rgb {
+    red: u8,
+    blue: u8,
+    green: u8,
+}
+
+fn read_rgb_from_csv(csv_data: &[u8]) -> Result<Rgb> {
+    let color: Rgb = csv::Reader::from_reader(csv_data)
+        .deserialize()
+        .nth(0)
+        .ok_or("Cannot deserialize the first CSV record")?
+        .chain_err(|| "Cannot deserialize RGB color")?;
+
+    Ok(color)
+}
+
+# type Hex = u32;
+#
+# fn rgb_to_hex(color: &Rgb) -> Hex {
+#     (color.red as Hex) << 16 | (color.blue as Hex) << 8 | (color.green as Hex)
+# }
+#
+fn run() -> Result<()> {
+    let csv = "red,blue,green
+102,256,204";
+
+    let rgb = read_rgb_from_csv(csv.as_bytes()).chain_err(|| "Cannot read CSV data")?;
+    let hex = rgb_to_hex(&rgb);
+    println!("{:?} to hexadecimal #{:X}", rgb, hex);
+
+    Ok(())
+}
+
+quick_main!(run);
```

Comment:
lets not use the `quick_main!` here as it will error out the application.
lets have a regular `main` in addition to `run` to showcase usage of
[`backtrace`](https://docs.rs/error-chain/0.11.0/error_chain/trait.ChainedError.html#tymethod.backtrace) in order to iterate over the backtrace